starboard/shared/posix/time_zone_get_name.cc

```diff
@@ -12,13 +12,59 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "starboard/common/log.h"
 #include "starboard/time_zone.h"
 
-#include <time.h>
+#define TZDEFAULT "/etc/localtime"
+#define TZZONEINFOTAIL "/zoneinfo/"
+#define isNonDigit(ch) (ch < '0' || '9' < ch)
+
+static bool isValidOlsonID(const char* id) {
```

Comment:
Let's make this function more robust and move it to starboard/common/. 
We should also call it from starboard/nplb/time_zone_get_name_test.cc to validate everyone's implmentation.

Ideally there would be some BNR form for the format similar to how RFC are defined, but the closes I thing to a spec that I found is the following:
```
Names normally have the form AREA/LOCATION, where AREA is a continent or ocean, and LOCATION is a specific location within the area. North and South America share the same area, 'America'. Typical names are 'Africa/Cairo', 'America/New_York', and 'Pacific/Honolulu'. Some names are further qualified to help avoid confusion; for example, 'America/Indiana/Petersburg' distinguishes Petersburg, Indiana from other Petersburgs in America.

Here are the general guidelines used for choosing timezone names, in decreasing order of importance:

Use only valid POSIX file name components (i.e., the parts of names other than '/'). Do not use the file name components '.' and '..'. Within a file name component, use only [ASCII](https://en.wikipedia.org/wiki/ASCII) letters, '.', '-' and '_'. Do not use digits, as that might create an ambiguity with [POSIX TZ strings](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03). A file name component must not exceed 14 characters or start with '-'. E.g., prefer America/Noronha to America/Fernando_de_Noronha. Exceptions: see the discussion of legacy names below.
```

Here is a link:
https://data.iana.org/time-zones/theory.html#naming

Reply:
Done. PTAL.

I wanted to use regex in the nplb but I got "regex.h is an unapproved C++11 header" when I'm trying to commit. So I changed to use gtest methods. I think this is sufficient to ensure IANA format.

The bigger problem is that some platforms can not pass this test, so I'll have to filter this new nplb test for now I think.

Reply:
OK, sg